tests/Doctrine/Tests/DBAL/Functional/Schema/MySqlSchemaManagerTest.php

```diff
@@ -319,4 +319,89 @@ public function testListFloatTypeColumns()
         $this->assertFalse($columns['col']->getUnsigned());
         $this->assertTrue($columns['col_unsigned']->getUnsigned());
     }
+
+    /**
+     * Test default null values (especially for mariadb 10.2.7+)
+     * @link https://github.com/doctrine/dbal/issues/2817
+     */
+    public function testNullColumnDefaultWithTable()
+    {
+
+        $table = new Table("test_column_defaults_with_table");
+        $table->addColumn('col0', 'integer', ['notnull' => false]);
+        $table->addColumn('col1', 'integer', ['notnull' => false, 'default' => null]);
+        $table->addColumn('col2', 'string', ['notnull' => false, 'default' => null]);
+        $table->addColumn('col3', 'string', ['notnull' => false, 'default' => 'NULL']);
+        $table->addColumn('col4', 'string', ['notnull' => false, 'default' => 'Hello world']);
+        $table->addColumn('col5', 'datetime', ['notnull' => false, 'default' => null]);
+        $table->addColumn('col6', 'decimal', ['scale' => 3, 'precision' => 6, 'notnull' => false, 'default' => -2.3]);
+        $table->addColumn('col7', 'date', ['notnull' => false, 'default' => '2012-12-12']);
+        $table->addColumn('col8', 'string', ['notnull' => true, 'default' => '']);
+        $table->addColumn('col9', 'integer', ['notnull' => false, 'default' => 0]);
+        $table->addColumn('col10', 'string', ['notnull' => false, 'default' => 'He"ll"o world']);
+
+        $this->_sm->dropAndCreateTable($table);
+
+        $onlineTable = $this->_sm->listTableDetails("test_column_defaults_with_table");
+        $this->assertNull($onlineTable->getColumn('col0')->getDefault());
```

Comment:
It's `self::assertNull`

Reply:
Nice, I never notices they assertNull, assertEquals... were static. I fixed. Thx